From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 13 Feb 2014 10:47:35 -0700
Subject: PCI/MSI: Fix leak of msi_attrs
Origin: https://git.kernel.org/linus/29237756477d5e777e39257d94af449fc66026c2
Bug-Debian: https://bugs.debian.org/748595

Coverity reported that I forgot to clean up some allocated memory on the
error path in populate_msi_sysfs(), so this patch fixes that.

Thanks to Dave Jones for pointing out where the error was, I obviously
can't read code this morning...

Found by Coverity (CID 1163317).

Fixes: 1c51b50c2995 ("PCI/MSI: Export MSI mode using attributes, not kobjects")
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Cc: Dave Jones <davej@redhat.com>
---
 drivers/pci/msi.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -521,6 +521,7 @@ error_attrs:
 		++count;
 		msi_attr = msi_attrs[count];
 	}
+	kfree(msi_attrs);
 	return ret;
 }
 
